jaxopt/_src/iterative_refinement.py

```diff
@@ -161,19 +159,30 @@ def update(self,
     state = IterativeRefinementState(
       iter_num=state.iter_num+1,
       error=error,
-      target_residuals=target_residuals,
-      init=None)
+      target_residuals=target_residuals)
 
     return base.OptStep(params, state)
 
   def run(self,
-          init_params: Optional[Any],
+          init_params,
           params_A: Any,
           b: Any,
           params_A_bar: Optional[Any] = None):
-
-    if init_params is None:
-      init_params = self.init_params(params_A, b, params_A_bar)
+    """Runs the iterative refinement.
+
+    Args:
+      init_params: ignored.
+      params_A: params for ``self.matvec_A``.
+      b: vector ``b`` in ``Ax=b``.
+      params_A_bar: optional parameters for ``matvec_A_bar``.
+    Returns:
+      (params, state), ``params = (primal_var, dual_var_eq, dual_var_ineq)``
+    """
+    del init_params
+    # TODO(lbethune): allow warm start when the linear_solve API reaches its final state. 
+    # Currently Jaxopt API uses init keyword but it is not exposed publicly.
```

Comment:
It is public but not documented (https://github.com/google/jaxopt/blob/main/jaxopt/linear_solve.py).